tests/unit-tests/test-class-sensei-data-cleaner.php

```diff
@@ -161,4 +189,33 @@ public function testSenseiPagesTrashed() {
 			$this->assertNotEquals( 'trash', get_post_status( $page_id ), 'Regular page should not be trashed' );
 		}
 	}
+
+	/**
+	 * Ensure the Sensei roles and caps are deleted.
+	 *
+	 * @covers Sensei_Data_Cleaner::cleanup_all
+	 * @covers Sensei_Data_Cleaner::cleanup_roles_and_caps
+	 */
+	public function testSenseiRolesAndCapsDeleted() {
+		Sensei_Data_Cleaner::cleanup_all();
+
+		// Refresh user info.
+		wp_cache_flush();
+
+		$regular_user = get_user_by( 'id', $this->regular_user_id );
+		$this->assertTrue( in_array( 'author', $regular_user->roles ), 'Author role should not be removed' );
+		$this->assertTrue( $regular_user->has_cap( 'edit_others_posts' ), 'Non-Sensei cap should not be removed from user' );
+		$this->assertFalse( $regular_user->has_cap( 'manage_sensei' ), 'Sensei cap should be removed from user' );
+
+		$teacher_user = get_user_by( 'id', $this->teacher_user_id );
```

Comment:
Again here. May as well make the entire `$teacher` variable available on creation.